homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -246,3 +593,22 @@ def update(self) -> None:
                 self._state = dt_util.parse_datetime(date_str).isoformat()
             else:
                 self._state = getattr(vehicle_last_trip, self._attribute)
+        elif self._service == SERVICE_ALL_TRIPS:
+            for attribute in [
+                "average_combined_consumption",
+                "average_electric_consumption",
+                "average_recuperation",
+                "chargecycle_range",
+                "total_electric_distance",
+            ]:
+                match = re.search(f"({attribute})_(.+)", self._attribute)
```

Comment:
This is a very expensive way of doing `if not self._attribute.starts_with(attribute): continue`